### PR TITLE
[FrameworkBundle][HttpClient] Add ThrottlingHttpClient

### LINKED ISSUE #19482: [FrameworkBundle][HttpClient] Add `ThrottlingHttpClient` to limit reque…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#53550
| PR author(s) | @HypeMC
| Merged in    | 7.1

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.1 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alamirault**
Fix #19482

### CODE REVIEW

**Review by 94noni — APPROVED** [review]
nice one

File: http_client.rst [hunk 1]
```diff
@@ -1474,6 +1474,40 @@ installed in your application::
 :class:`Symfony\\Component\\HttpClient\\CachingHttpClient` accepts a third argument
 to set the options of the :class:`Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache`.
 
+Limits Number of Requests
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
Limit the Number of Requests
```

File: http_client.rst [hunk 2]
```diff
@@ -1474,6 +1474,40 @@ installed in your application::
...
+-------------------------
+
+This component provides a :class:`Symfony\\Component\\HttpClient\\ThrottlingHttpClient`
+decorator that allows to limits the number of requests within a certain period.
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
decorator that allows to limit the number of requests within a certain period.
```

File: http_client.rst [hunk 3]
```diff
@@ -1474,6 +1474,40 @@ installed in your application::
...
+    $client = HttpClient::create();
+    $client = new ThrottlingHttpClient($client, $rateLimiter);
+
+    $requests = [];
+    for ($i = 0; $i < 100; $i++) {
+        $requests[] = $client->request('GET', 'https://example.com');
+    }
+
+    foreach ($requests as $request) {
+        // Depending of rate limiting policy, calls will be delayed
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
        // Depending on rate limiting policy, calls will be delayed
```

**Review by OskarStark — APPROVED** [review]

File: http_client.rst [hunk 4]
```diff
@@ -1474,6 +1474,40 @@ installed in your application::
...
 
+Limit the Number of Requests
+----------------------------
+
+This component provides a :class:`Symfony\\Component\\HttpClient\\ThrottlingHttpClient`
+decorator that allows to limit the number of requests within a certain period.
+
+The implementation leverages the
+:class:`Symfony\\Component\\RateLimiter\\LimiterInterface` class under the hood
+so that the :doc:`Rate Limiter component </rate_limiter>` needs to be
```
Comment (smnandre) [inline http_client.rst:1485]:
```suggestion
so the :doc:`Rate Limiter component </rate_limiter>` needs to be
```

**Comment by javiereguiluz** [comment]
Thanks Antoine!

File: http_client.rst (hunk 4 above)
Comment (javiereguiluz) [reply http_client.rst:1485]: Changed while merging. Thanks!

**Pull request merged**